gammapy/background/energy_offset_array.py

```diff
@@ -71,7 +95,11 @@ def _fill_one_event_list(self, events):
         ev_cube_hist, ev_cube_edges = np.histogramdd(ev_cube_array,
                                                      [self.energy.value,
                                                       self.offset.value])
-        return ev_cube_hist
+        energy_bin = np.diff(ev_cube_edges[0])
+        solid_angle_bin = np.pi * np.diff(ev_cube_edges[1] ** 2)
+        solid_angle_tab, energy_tab = np.meshgrid(solid_angle_bin, energy_bin)
+        bkg_rate = ev_cube_hist / (solid_angle_tab * energy_tab)
```

Comment:
My suggestion would be that you keep this pull request limited to the I/O and make a separate one with the new class and rate computation.
That way this one can be finished relatively quickly and review is easier.

I'll review the I/O methods later today ... for sure tests are needed.
Here's an example of a test that writes an object to FITS, reads it back, and checks that the content is OK:
https://github.com/gammapy/gammapy/blob/1fde3292a5f2f66da0be758249cb6f63fc478400/gammapy/scripts/tests/test_image_bin.py#L11

A similar test should be added here as well.
Make an example array, write, read back, assert something, e.g. an array entry value or first and last axis bin value.

Does this make sense, or should I describe in more detail how the I/O test should be implemented?